lib/elixir/src/elixir_parser.yrl

```diff
@@ -1217,9 +1217,13 @@ warn_pipe(_Token, _) ->
 %% TODO: Make this an error on v2.0
 warn_nested_no_parens_keyword(Key, Value) ->
   {line, Line} = lists:keyfind(line, 1, ?meta(Value)),
+  MaybeKeyword = case Key of
+    Atom when is_atom(Atom) ->  "\"" ++ atom_to_list(Key) ++ ":\" ";
+    _ -> ""
```

Comment:
The literal_encoder might be returning an arbitrary term, in which case we probably can't use it for the warning.

Reply:
Great find!

So my suggestion is to keep the warning altogether if the key is not an atom, WDYT? We already explicitly avoid some warnings when formatting anyway.

Reply:
I thought about it as well, the reason I went with this approach is because `literal_encoder` is part of the public API of [`Code.string_to_quoted/2`](https://hexdocs.pm/elixir/Code.html#string_to_quoted/2), so there might theoretically be different use cases than formatting?
WDYT? I'm fine with both approaches.

Reply:
Let's skip the warning cause I am not convinced showing the warning without knowing the key is that helpful. :) Thanks!

Reply:
Done in [be2ea24](https://github.com/elixir-lang/elixir/pull/12274/commits/be2ea2449d9abbc550dd9e55daa2063f890ab7bf)